### PR TITLE
Implemented CloudSync Feature

### PR DESCRIPTION
**Pull request by anuragm339**
his PR contains implementation of cloud sync bootstrap type

### CODE REVIEW

File: registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java [hunk 1]
```diff
@@ -65,6 +70,18 @@ public BootstrapType requiresBootstrap(Node node) throws SQLException {
         }
     }
 
+    @Override
+    public BootstrapType getLatestBootStrapType(Node node) throws SQLException{
+        try (Connection connection = getConnection()) {
+            BootstrapType bootstrapType = readLasBootstrapType(node.getHost(), connection);
+            return bootstrapType;
+        } catch (SQLException exception) {
+            LOG.error("read requiresBootstrap()", "read a node request", exception);
```
Comment (ryzhovart) [inline registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java:79]: Is there reading a node request or getting the latest bootstrap type? Please update logging.

File: registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java [hunk 2]
```diff
@@ -76,12 +93,23 @@ private BootstrapType readBootstrapType(String hostId, Connection connection) th
         }
     }
 
+    private BootstrapType readLasBootstrapType(String hostId, Connection connection) throws SQLException {
```
Comment (ryzhovart) [inline registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java:96]: typo in message name

File: registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java [hunk 3]
```diff
@@ -76,12 +93,23 @@ private BootstrapType readBootstrapType(String hostId, Connection connection) th
...
+        long start = System.currentTimeMillis();
+        try (PreparedStatement statement = connection.prepareStatement(QUERY_TO_GET_LAST_BOOTSTRAP)) {
+            statement.setString(1, hostId);
+            return getBootstrapType(statement.executeQuery());
+        } finally {
+            long end = System.currentTimeMillis();
+            LOG.info("read:time readBootstrapType()", Long.toString(end - start));
```
Comment (ryzhovart) [inline registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java:103]: where clause should contain methodName()

File: registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java (hunk 1 above)
Comment (anuragm339) [reply registry-http-server/src/main/java/com/tesco/aqueduct/registry/postgres/PostgreSQLNodeRequestStorage.java:79]: We are reading the last booststrap type I will update the naming

**Review by NPhalgunaRao — APPROVED** [review]

**Review by ryzhovart — COMMENTED** [review]
LGTM

**Pull request merged**